### PR TITLE
[SDPA-1887] Add webform Price To field custom validation, and custom properties

### PR DESCRIPTION
**Pull request by enzolutions**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-1887

Related PRs: 
* Content PR:  

### Changed

1. Add webform Price To field custom validation
2. Add custom properties to Price To field custom to be delivered via JSONAPI

### Screenshots
screen shot 2019-02-19 at 11 20 45 am

### CODE REVIEW

File: src/Validate/PriceConstraint.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   * @param array $form
+   *   The complete form structure.
+   */
+  public static function validate(array &$element, FormStateInterface $formState, array &$form) {
+    $error = FALSE;
+    $webformKey = $element['#webform_key'];
+    $price_from_value = $formState->getValue('price_from');
+    $price_to_value = $formState->getValue($webformKey);
+
+    error_log("Price From: " . $price_from_value, 0);
```
Comment (sonnykt) [inline src/Validate/PriceConstraint.php]: @enzolutions why do you use `error_log` here?

File: src/Validate/PriceConstraint.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    if ($error) {
+      if (isset($element['#title'])) {
+        $tArgs = [
+          '%name' => empty($element['#title']) ? $element['#parents'][0] : $element['#title'],
+          '%value' => $price_to_value,
+        ];
+        $formState->setError(
+          $element,
+          t('The %name value "%value" should be greater or equal to "Price From" value. Please use a different amount.', $tArgs)
```
Comment (sonnykt) [inline src/Validate/PriceConstraint.php]: @enzolutions use dependency injection for language translation instead of direct `t()`.

**Review by sonnykt — CHANGES_REQUESTED** [review]

File: src/Validate/PriceConstraint.php (hunk 1 above)
Comment (enzolutions) [reply src/Validate/PriceConstraint.php]: Sorry that was my frustration with CI , I will remove it

File: src/Validate/PriceConstraint.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    if ($error) {
+      if (isset($element['#title'])) {
+        $tArgs = [
+          '%name' => empty($element['#title']) ? $element['#parents'][0] : $element['#title'],
+          '%value' => $price_to_value,
+        ];
+        $formState->setError(
+          $element,
+          new TranslatableMarkup('The %name value "%value" should be greater or equal to "Price From" value. Please use a different amount.', $tArgs)
```
Comment (sonnykt) [inline src/Validate/PriceConstraint.php:46]: @enzolutions both `t()` and `TranslatableMarkup` should not be used directly. You should instead use dependency injection to call the translation service. See https://github.com/sonnykt/module_permissions/blob/8.x-1.x/src/UninstallValidator.php

Secondly, why do we need a new class just for form element validation? Don't just follow the [webform article](https://www.drupal.org/docs/8/modules/webform/webform-cookbook/how-to-add-custom-validation-to-a-webform-element) as it's overkilled. You only need a new class if you want to define a [constraint validator](https://www.drupal.org/docs/8/api/entity-validation-api/providing-a-custom-validation-constraint) and your constraint can be used elsewhere. And standard validate callback is enough and you don't have to deal with dependency injection.

**Review by sonnykt — CHANGES_REQUESTED** [review]

File: src/Validate/PriceConstraint.php (hunk 3 above)
Comment (enzolutions) [reply src/Validate/PriceConstraint.php:46]: OK I will use the Trait.

About the webform I just follow the official documentation for Webform https://www.drupal.org/docs/8/modules/webform/webform-cookbook/how-to-add-custom-validation-to-a-webform-element

**Review by sonnykt — APPROVED** [review]

File: src/Validate/PriceConstraint.php (hunk 3 above)
Comment (anthony-malkoun) [reply src/Validate/PriceConstraint.php:46]: @sonnykt Just picking this up now that Enzo has left. It doesn't look like this has progressed with the changes you requested. Are there changes that didn't get pushed to this branch that I don't know about? I agree with injecting `TranslatableMarkup` (good pickup!) but could be convinced either way re: procedural versus OO. What is the best way to get this ticket completed?

**Pull request merged**